app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt

```diff
@@ -59,7 +58,7 @@ class HistoryFragment : Fragment(), CoroutineScope, BackHandler {
         Navigation.findNavController(requireActivity(), R.id.container).apply {
             navigate(
                 HistoryFragmentDirections.actionGlobalBrowser(null,
-                    (activity as HomeActivity).browsingModeManager.isPrivate),
+                    requireComponents.core.sessionManager.selectedSession!!.private),
```

Comment:
I would never assume there's a selected session. It's definitely not true on custom tabs. It's likely not true when the user first enters the app.

Reply:
Also: If you really really are sure (maybe never in Fenix?) then you can also use [SessionManager.selectedSessionOrThrow](https://mozac.org/api/mozilla.components.browser.session/-session-manager/selected-session-or-throw.html)